Dockerfile

```diff
@@ -45,21 +48,22 @@ RUN set -ex\
 FROM base AS build-python
 ENV PYTHONDONTWRITEBYTECODE 1
 RUN set -ex\
-	; dnf -y -q --setopt=tsflags=nodocs --setopt=skip_missing_names_on_install=False install\
-		gcc-c++\
-		git\
-		openldap-devel\
-		python39-devel\
-		libffi-devel\
+	; microdnf -y --setopt=tsflags=nodocs install \
+		gcc-c++ \
+		git \
+		openldap-devel \
+		python39-devel \
+		libffi-devel \
         openssl-devel \
         diffutils \
         file \
         make \
         libjpeg-turbo \
         libjpeg-turbo-devel \
-		wget\
-	; dnf -y -q clean all
+		wget \
+	; microdnf -y clean all
 WORKDIR /build
+RUN python3 -m ensurepip --upgrade
```

Comment:
Downstream build system doesn't allow us to download unpinned versions. Most likely it'll be removed when we reconcile upstream and downstream Dockerfile (ideally we should be able to generate downstream Dockerfile with `sed`)

Reply:
Isn't ensurepip part of python and hence the pip version is aligned to the python version?

Reply:
TIL (from the front documentation page):

> This module (ensurepip) does not access the internet. All of the components needed to bootstrap pip are included as internal parts of the package.